lib/calendar.ts

```diff
@@ -1794,7 +1852,7 @@ const nonIsoGeneralImpl = {
     cache.setObject(result);
     return result;
   },
-  yearMonthFromFields(fields, options, calendar) {
+  yearMonthFromFields(this: typeof nonIsoGeneralImpl & { helper: NonIsoHelperBase }, fields, options, calendar) {
```

Comment:
Would it make sense to abstract this type into a global type declaration to avoid repeating `typeof nonIsoGeneralImpl & { helper: NonIsoHelperBase }` in lot of places?

Reply:
When I tried this I got an error: `Type alias 'NonIsoGeneralImplThis' circularly references itself.ts(2456)`.

Example:
```
type NonIsoGeneralImplThis = typeof nonIsoGeneralImpl & { helper: NonIsoHelperBase };

... 

const nonIsoGeneralImpl = {
  dateFromFields(this: NonIsoGeneralImplThis, fields, options, calendar) {
    const overflow = ES.ToTemporalOverflow(options);
```

Reply:
Ahhh, OK. Good thing for a future black-belt TS contributor to figure out. Not req'd now.

BTW, I apologize for the TS-unfriendly structure of this code. I wrote it this way to simplify adapting the code from the previous non-ISO-calendar implementation which used a similar class-like-but-not-really structure for these singletons. If I were starting all over again I'd probably do it differently. Sorry!